og_ui/src/Controller/SubscriptionController.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\og_ui\Controller\SubscriptionController.
+ */
+
+namespace Drupal\og_ui\Controller;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Field\FieldStorageDefinitionInterface;
+use Drupal\Core\Url;
+use Drupal\user\Entity\User;
+use Drupal\user\EntityOwnerInterface;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Drupal\og\Og;
+use Drupal\og\OgAccess;
+
+/**
+ * Controller for OG subscription routes.
+ */
+class SubscriptionController extends ControllerBase {
+
+  /**
+   * @param string $entity_type_id
+   * @param string|int $entity_id
+   *
+   * @return mixed
+   */
+  public function subscribe(Request $request, $entity_type_id, $entity_id) {
+    // @todo We don't need to re-validate the entity type and entity group here,
+    // as it's already been done in the access check?
+    $entity_storage = $this->entityTypeManager()->getStorage($entity_type_id);
+    $entity_access = $this->entityTypeManager()->getAccessControlHandler($entity_type_id);
+    $group = $entity_storage->load($entity_id);
+
+    $account = User::load($this->currentUser()->id());
+    $field_name = $request->query->get('field_name');
```

Comment:
No not really :) Seems like we could just use a field_name slug with a default value set.